### PR TITLE
Improve job cache

### PR DESCRIPTION
**Pull request by shinytang6**
Just found some minor optimizations when reading the code :)

### CODE REVIEW

File: pkg/controllers/cache/cache.go [hunk 1]
```diff
@@ -111,6 +111,10 @@ func (jc *jobCache) GetStatus(key string) (*v1alpha1.JobStatus, error) {
 		return nil, fmt.Errorf("failed to find job <%s>", key)
 	}
 
+	if job.Job == nil {
+		return nil, fmt.Errorf("job <%s> is not ready", key)
```
Comment (hzxuzhonghu) [inline pkg/controllers/cache/cache.go:115]: nit: this function is used only by unit test

**Review by hzxuzhonghu — COMMENTED** [review]
/lgtm
/approve

**Comment by william-wang** [comment]
@shinytang6 please fix the build failure.

**Comment by shinytang6** [comment]
@hzxuzhonghu @william-wang Can y help me with that ? l think CI failure is not caused by my changes.

**Review by william-wang — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
@shinytang6 Not sure why the travis ci always went wrong, maybe you can file a new pr

**Comment by shinytang6** [comment]
@hzxuzhonghu lt seems still not work, ci still fails :no_mouth:

**Comment by hzxuzhonghu** [comment]
Yes i know.  Looks like a bug of travis or go.  FYI https://github.com/volcano-sh/volcano/pull/1085

**Comment by hzxuzhonghu** [comment]
It seems you need to rebase and so that the new travis config can take effect

**Comment by shinytang6** [comment]
> It seems you need to rebase and so that the new travis config can take effect

Sure, but because of the problem reported here #1076, it is very difficult for me to synchronize upstream with my local master, I have to fork the project again, so please refer to #1084 and merge that one :)

**Pull request closed without merging**